lib/cdo/aws/rds.rb

```diff
@@ -0,0 +1,169 @@
+require_relative '../../../deployment'
+require 'aws-sdk-rds'
+
+module Cdo
+  class RDS
+    # Create an RDS Aurora MySQL cluster cloned from a specified source cluster.  Make and use copies of the source
+    # Parameter Groups to avoid inadvertent configuration changes to the source cluster.  Unfortunately, it isn't
+    # possible to use a CloudFormation template to provision the cluster due to an issue where CloudFormation ignores
+    # the DBClusterParameterGroup property:
+    # @param source_cluster_id [String] DB cluster id of the cluster to clone.  Defaults to current environment's cluster.
+    # @param clone_cluster_id [String] DB cluster id to assign to clone.  Defaults to source cluster id + "-clone"
+    # @param instance_type [String] RDS DB Instance Type
+    # @param max_attempts [Integer] Number of times to check whether task has completed successfully before failing.
+    # @param delay [Integer] Number of seconds to wait between checking task status.
+    def self.clone_cluster(
+      source_cluster_id: CDO.db_cluster_id,
+      clone_cluster_id: "#{source_cluster_id}-clone",
+      instance_type: 'db.r4.large',
+      max_attempts: 30,  # It takes ~15 minutes to clone the production cluster, so default to 30 minutes.
+      delay: 60
+    )
+      clone_instance_id = clone_cluster_id + "-0"
+      rds_client = Aws::RDS::Client.new
+      begin
+        CDO.log.info "Creating clone of database cluster - #{source_cluster_id}"
+        source_cluster = rds_client.describe_db_clusters({db_cluster_identifier: source_cluster_id}).db_clusters.first
+
+        copy_source_cluster_parameter_group = rds_client.copy_db_cluster_parameter_group(
+          {
+            source_db_cluster_parameter_group_identifier: source_cluster[:db_cluster_parameter_group],
+            target_db_cluster_parameter_group_description: "#{clone_cluster_id}-auroraclusterdbparameters",
+            target_db_cluster_parameter_group_identifier: "#{clone_cluster_id}-auroraclusterdbparameters",
```

Comment:
Because I decided not to combine these 2 methods into a `Cluster` Class, I think a constant wouldn't work here? But I did update the `clone_cluster` method to use variables instead of repeating the same string.